arviz/stats/stats.py

```diff
@@ -45,7 +43,14 @@
 
 
 def compare(
-    dataset_dict, ic=None, method="stacking", b_samples=1000, alpha=1, seed=None, scale=None
+    dataset_dict: Optional[Dict[str, InferenceData]],
```

Comment:
I don't think this is optional, and I think we should use `Mapping` instead of dict for inputs, similarly to List vs Sequence

Reply:
That looks right to me. I will change it. I can't recall why I thought `Optional` was right.